### ISSUE TITLE
INSTALL_FAILED_CONFLICTING_PROVIDER

### ISSUE THREAD

**Issue by bubudge** [kloder-games/godot-admob#48]
**OS target (Android/iOS):**
Android 7.0

**Godot version:**
v3.0.6-stable

**Issue description:**
I compiled custom packages(debug and release) from godot and godot-admob sources.
And I build my application packages for android with the custom packages(debug and release).
Both packages worked well with ads in my tablet.
The next, I build my another application with the same custom packages(debug and release).
When I was going to install the app in my tablet, it ended with error messages below.
-------
[INSTALL_FAILED_CONFLICTING_PROVIDER: Package couldn't be installed in /data/app/net.bubudge.capital-1: Can't install because provider name com.godot.game.mobileadsinitprovider (in package net.bubudge.capital) is already used by net.bubudge.uspresidents]
-------
So I uninstalled my first application from my tablet, and I tried to install my second application.
It worked well with ads in my tablet.

I don't know how to change the provider name and how to fix this problem.
Could you help me?

**Comment by Shin-NiL** [comment]
You must change your app package name, to do so, go to Project -> Export -> Android -> Package -> Unique Name. Each app must have a unique package name.

Just to be clear, it's not a module issue.

**Comment by bubudge** [comment]
Thank you for your reply.
I've already give all packages unique names.
Now I realized this isn't a module issue.

**Comment by bubudge** [comment]
I don't know how to solve this problem, but I found custom packages made by another man.
This worked good for me.
https://www.youtube.com/watch?v=jaaZh__YwlM&index=24&list=LLirCbduARnb4_wliNTN_GZQ&t=522s

**Comment by bubudge** [comment]
I tried a new commit and it worked well.
Thank you very much.

**Comment by phaitonican** [comment]
ye works fine for me too with new master!
Ty

**Comment by sadat-s-ahmed** [comment]
I am having the same issue. I changed the unique package name but still same issue. I;m using ad apk 17.1.3 and godot 3.1beta. The build is fine, but it conflicts with my other game projects. I am using the latest master. Any suggestion will be greatly appreciated

I am also confused as to where this package name is being generated. Would be great if you could point me at it 

com.godot.game.mobileadsinitprovider

**Comment by Shin-NiL** [comment]
Take a look at issue #62

**Comment by sadat-s-ahmed** [comment]
@Shin-NiL  Thanks for the pointer. Got it running finally.

I managed to get it running using ads 17.1.3
ndk 19b and 28.0.3 build tool

The main issue was that I had to build the export templates without the admob module using the support library 28, move it to templates, then recompile the ad module again but this time removing the support library and linking this new build to the game.

Haven't been having these issues in the older build tools so this was a bit of an annoyance. Thanks.
